lib/uploader/uploader.ts

```diff
@@ -491,8 +492,16 @@ export class Uploader {
 			if (!disabledChunkUpload) {
 				logger.debug('Initializing chunked upload', { file, upload })
 
+				let chunksWorkspace: string
+				if (this._isPublic) {
+					const shareToken = root!.split('/').at(-1)!
```

Comment:
This is quite dirty, instead just use `getShareToken` from `@nextcloud/sharing`?